### ISSUE TITLE
`no-useless-return` removes return statements in unreachable code

### ISSUE THREAD

**Issue by benjie** [eslint/eslint#11647]
**Tell us about your environment**

* **ESLint Version:** v5.16.0 (via the ESLint.org/demo website)
* **Node Version:** ESLint.org/demo website
* **npm Version:** ESLint.org/demo website

**What parser (default, Babel-ESLint, etc.) are you using?**

ESLint.org/demo default

**Please show your full configuration:**

[This eslint.org/demo configuration](https://eslint.org/demo/#eyJ0ZXh0IjoiZnVuY3Rpb24gZm9vKGFyZykge1xuICB0aHJvdyBuZXcgRXJyb3IoXCJEZWJ1Z2dpbmcuLi5cIik7XG5cbiAgaWYgKCFhcmcpIHtcbiAgICByZXR1cm47XG4gIH1cblxuICBjb25zb2xlLmxvZyhhcmcpO1xufSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6NSwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6eyJuby11c2VsZXNzLXJldHVybiI6Mn0sImVudiI6e319fQ==) (configured with just the `no-useless-return` rule)

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
function foo(arg) {
  throw new Error("Debugging...");

  if (!arg) {
    return; // < This return incorrectly gets removed by auto-fix
  }

  console.log(arg);
}
```

Using eslint.org/demo link above

I added a `throw` statement to the top of my function to aid debugging, and the critical return statement from later in the code was dropped which I only discovered after removing the `throw` statement and strange errors occurring.

**What did you expect to happen?**

`return;` statement should be flagged in unreachable code if and only if it would be flagged in equivalent reachable code. (At least that's how I read [the comment in the source code](https://github.com/eslint/eslint/blob/8684f4646da33bfe81e8f7f8c2d1af8b31065564/lib/rules/no-useless-return.js#L102-L108).)

If we were to remove the `throw` line, the `return;` statement would not be flagged; therefore it should not be flagged with the `throw` statement present.

**What actually happened? Please include the actual, raw output from ESLint.**

`5:5 Unnecessary return statement.` and fixed code removes the `return` statement. (See eslint.org/demo link)

**Are you willing to submit a pull request to fix this bug?**

This is currently beyond my ESLint skillset.
